app/scripts/views/marketing_snippet.js

```diff
@@ -61,6 +105,33 @@ define(function (require, exports, module) {
       const isMobileIosFirefox = /FxiOS/.test(ua);
 
       return isMobileAndroidFirefox || isTabletAndroidFirefox || isMobileIosFirefox;
+    },
+
+    _isIos () {
+      const plat = this.window.navigator.platform;
+
+      return /i(Phone|Pad|Pod)/.test(plat);
+    },
+
+    _isAndroid () {
+      return /android/i.test(this.window.navigator.userAgent);
```

Comment:
Yeah, I'm already working on it in the complete-on-fennec branch! Maybe I'll separate that work out into its own PR so it too is simpler to review.